pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -309,8 +300,8 @@ func TestCollectObservationLog(t *testing.T) {
 	for name, test := range testCases {
 		t.Run(name, func(t *testing.T) {
 			actual, err := CollectObservationLog(test.filePath, test.metrics, test.filters, test.fileFormat)
-			if (err != nil) != test.err {
-				t.Errorf("\nGOT: \n%v\nWANT: %v\n", err, test.err)
+			if err != nil && err.Error() != test.errCause {
+				t.Errorf("Unexpected error (-want,+got):\n%s", cmp.Diff(test.errCause, err.Error()))
```

Comment:
We must not compare error strings.
Please refer to this example: https://github.com/kubeflow/katib/blob/6f372f68089c0a01d2c06e98489557a88e5a7183/pkg/util/v1beta1/katibconfig/config_test.go#L42

Reply:
Okay, shall I define those errors in test cases directly?

Reply:
Checking here how to define and use errors: 

https://github.com/kubeflow/katib/blob/6f372f68089c0a01d2c06e98489557a88e5a7183/pkg/util/v1beta1/katibconfig/config.go#L38-L42

https://github.com/kubeflow/katib/blob/6f372f68089c0a01d2c06e98489557a88e5a7183/pkg/util/v1beta1/katibconfig/config.go#L144

Reply:
Do you mean that I need to modify the error msg returned in `file-metricscollector.go` and compare errors in `file-metricscollector_test.go`?

Reply:
Yes, I meant it.

Reply:
AFAIK, it will change the current error handling code in `file-metricscollector.go`

Reply:
Okay, I'll do it later.